clio-util/src/main/scala/org/broadinstitute/clio/util/json/ModelAutoDerivation.scala

```diff
@@ -61,6 +63,16 @@ trait ModelAutoDerivation extends AutoDerivation {
     }
   }
 
+  implicit val encodeUpsertId: Encoder[UpsertId] =
+    Encoder.encodeString contramap [UpsertId] { _.id }
+
+  implicit val decodeUpsertId: Decoder[UpsertId] = (c: HCursor) =>
+    for {
+      upsertId <- c.as[String]
+    } yield {
+      UpsertId(upsertId)
```

Comment:
Are there any properties about the `upsertId` that this should check? Length would be an easy one.

Reply:
Yeah, it would make sense to do this, inside `UpsertId` or `UpsertIdGenerator`.